.github/workflows/build_and_test.yml

```diff
@@ -27,11 +27,12 @@ jobs:
       style_checker_image_name: "ghcr.io/citusdata/stylechecker"
       style_checker_tools_version: "0.8.18"
       sql_snapshot_pg_version: "16.3"
-      image_suffix: "-v13fd57c"
+      image_suffix: "-dev-e5b0348"
       pg14_version: '{ "major": "14", "full": "14.12" }'
       pg15_version: '{ "major": "15", "full": "15.7" }'
       pg16_version: '{ "major": "16", "full": "16.3" }'
-      upgrade_pg_versions: "14.12-15.7-16.3"
+      pg17_version: '{ "major": "17", "full": "17.0" }'
+      upgrade_pg_versions: "14.12-15.7-16.3-17.0"
```

Comment:
Should this matrix include 17 for upgrade testing?

https://github.com/citusdata/citus/blob/f6959715dc49404510f2f3340b9719b33ddd152b/.github/workflows/build_and_test.yml#L303

Reply:
Yes, but not in this PR. As I wrote in the PR description, these changes will not be merged for now. I have only added the pg17-build test to help with PR validation.